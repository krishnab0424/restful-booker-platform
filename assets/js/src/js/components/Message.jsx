import React from 'react';
import ReactModal from 'react-modal';
import { API } from '../libs/Api';

export default class Message extends React.Component {

    constructor(){
        super();

        this.state = {
            name : "",
            email : "",
            phone : "",
            subject : "",
            description : "",
        }
    }

    componentDidMount(){
        API.putMessageRead(this.props.messageId);
        API.getMessage(this.props.messageId, this);
    }

    render(){
        return <ReactModal 
                    isOpen={true}
                    contentLabel="onRequestClose Example"
                    className="message-modal" >
                    
                    <div className="form-row">
                        <div className="col-10">
                            <p><span>From: </span>{this.state.name}</p>
                        </div>
                        <div className="col-2">
                            <p><span>Phone: </span>{this.state.phone}</p>
                        </div>
                    </div>
                    <div className="form-row">
                        <div className="col-12">
                            <p><span>Email: </span>{this.state.email}</p>
                        </div>
                    </div>
                    <div className="form-row">
                        <div className="col-12">
                            <p><span>{this.state.subject}</span></p>
                        </div>
                    </div>
                    <div className="form-row">
                        <div className="col-12">
                            <p>{this.state.description}</p>
                        </div>
                    </div>
                    <div className="form-row">
                        <div className="col-12">
                            <button className="btn btn-outline-primary" onClick={() => this.props.closeMessage()}>Close</button>
                        </div>
                    </div>
                </ReactModal>
    }

}