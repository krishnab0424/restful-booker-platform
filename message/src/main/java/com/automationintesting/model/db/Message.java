package com.automationintesting.model.db;

import com.fasterxml.jackson.annotation.JsonProperty;

import javax.persistence.Entity;
import javax.validation.constraints.*;
import java.sql.ResultSet;
import java.sql.SQLException;

@Entity
public class Message {

    @JsonProperty
    private int messageid;

    @JsonProperty
    @NotNull(message = "Name must be set")
    @NotBlank
    private String name;

    @JsonProperty
    @Email
    @NotNull(message = "Email must be set")
    @NotBlank
    private String email;

    @JsonProperty
    @Size(min = 11, max = 21)
    @NotNull(message = "Phone must be set")
    @NotBlank
    private String phone;

    @JsonProperty
    @Size(min = 5, max = 100)
    @NotNull(message = "Subject must be set")
    @NotBlank
    private String subject;

    @JsonProperty
    @Size(min = 20, max = 2000)
    @NotNull(message = "Description must be set")
    @NotBlank
    private String description;

    public Message() {
    }

    public Message(String name, String email, String phone, String subject, String description) {
        this.name = name;
        this.email = email;
        this.phone = phone;
        this.subject = subject;
        this.description = description;
    }

    public Message(ResultSet result) throws SQLException {
        this.messageid = result.getInt("messageid");
        this.name = result.getString("name");
        this.email = result.getString("email");
        this.phone = result.getString("phone");
        this.subject = result.getString("subject");
        this.description = result.getString("description");
    }

    public int getMessageid() {
        return messageid;
    }

    public String getName() {
        return name;
    }

    public String getEmail() {
        return email;
    }

    public String getPhone() {
        return phone;
    }

    public String getSubject() {
        return subject;
    }

    public String getDescription() {
        return description;
    }

    public void setMessageid(int messageid) {
        this.messageid = messageid;
    }

    public void setName(String name) {
        this.name = name;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public void setPhone(String phone) {
        this.phone = phone;
    }

    public void setSubject(String subject) {
        this.subject = subject;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    @Override
    public String toString() {
        return "Message{" +
                "messageid=" + messageid +
                ", name='" + name + '\'' +
                ", email='" + email + '\'' +
                ", phone='" + phone + '\'' +
                ", subject='" + subject + '\'' +
                ", description='" + description + '\'' +
                '}';
    }
}
