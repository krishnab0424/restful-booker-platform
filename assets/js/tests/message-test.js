import React from 'react';
import Message from '../src/js/components/Message.jsx';
import nock from 'nock';
import { waitForState } from 'enzyme-async-helpers';

beforeAll(() => {
    nock('http://localhost')
        .persist()
        .get('/message/1')
        .reply(200, {
                "name" : "Mark Winteringham",
                "email" : "mark@mwtestconsultancy.co.uk",
                "phone" : "01821 912812",
                "subject" : "Subject description here",
                "description" : "Lorem ipsum dolores est"
            }
        );

    nock('http://localhost')
        .persist()
        .put('/message/1/read')
        .reply(200);
});

test('Message popup is populated with details', async () => {
    const messageComponent = shallow(<Message messageId={1} />);

    await waitForState(messageComponent, state => state.name.length > 0);

    expect(messageComponent).toMatchSnapshot();
});