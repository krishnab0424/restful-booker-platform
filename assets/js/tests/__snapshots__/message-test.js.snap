// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message popup is populated with details 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="message-modal"
  closeTimeoutMS={0}
  contentLabel="onRequestClose Example"
  isOpen={true}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <div
    className="form-row"
  >
    <div
      className="col-10"
    >
      <p>
        <span>
          From: 
        </span>
        Mark Winteringham
      </p>
    </div>
    <div
      className="col-2"
    >
      <p>
        <span>
          Phone: 
        </span>
        01821 912812
      </p>
    </div>
  </div>
  <div
    className="form-row"
  >
    <div
      className="col-12"
    >
      <p>
        <span>
          Email: 
        </span>
        mark@mwtestconsultancy.co.uk
      </p>
    </div>
  </div>
  <div
    className="form-row"
  >
    <div
      className="col-12"
    >
      <p>
        <span>
          Subject description here
        </span>
      </p>
    </div>
  </div>
  <div
    className="form-row"
  >
    <div
      className="col-12"
    >
      <p>
        Lorem ipsum dolores est
      </p>
    </div>
  </div>
  <div
    className="form-row"
  >
    <div
      className="col-12"
    >
      <button
        className="btn btn-outline-primary"
        onClick={[Function]}
      >
        Close
      </button>
    </div>
  </div>
</Modal>
`;
